ts/Extensions/ExportData.ts

```diff
@@ -466,6 +466,9 @@ addEvent(Chart, 'render', function (): void {
         !this.dataTableDiv
     ) {
         this.viewData();
+    } else if (this.dataTableDiv) {
+        // Data table was not re-rendered after a data update, #14320.
+        AST.setElementHTML(this.dataTableDiv, this.getTable());
```

Comment:
This seems a bit out of place, especially since we're otherwise doing the `setElementHTML` in the `toggleData` function. I found that the table successfully updates if I just remove the `!this.dataTableDiv` part of the condition:

```ts
addEvent(Chart, 'render', function (): void {
    if (
        this.options &&
        this.options.exporting &&
        this.options.exporting.showTable &&
        !this.options.chart.forExport
    ) {
        this.viewData();
    }
});
```